### PR TITLE
[Task] desensitization of main fields in logs #3861 #3927

### PR DESCRIPTION
**Pull request by wanghuan0606**
[Task] desensitization of main fields in logs https://github.com/apache/shenyu/issues/3861

1.I mask logging data according keyword like uri,name,id et. 
2.And i use regex to match keyword.
3.I add two new plugin handler keyword and maskType,and three dictionary value ,so user can add keyword and choose maskType when they make the logging rules,like this:
截屏2022-09-19 03 19 24
4.I provide two mask types include charReplace and MD5,then user can extend their own mask types.Only need implements DataMaskInterface and create new dictionary that dictionary type is "maskType" and dictionary value is their function.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/body/LoggingServerHttpResponse.java [hunk 1]
```diff
@@ -250,4 +273,107 @@ private String getUpstreamIpFromHttpDomain() {
...
+        }
+        if (keyWordMatch.matches(GenericLoggingConstant.TRACE_ID)
+                && StringUtils.isNotBlank(logInfo.getTraceId())) {
+            logInfo.setTraceId(dataMaskInterface.mask(logInfo.getTraceId()));
+        }
+        if (keyWordMatch.matches(GenericLoggingConstant.REQUEST_HEADER)
+                && StringUtils.isNotBlank(logInfo.getRequestHeader())) {
+            logInfo.setRequestHeader(dataMaskInterface.mask(logInfo.getRequestHeader()));
+        }
+        if (keyWordMatch.matches(GenericLoggingConstant.RESPONSE_HEADER)
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/body/LoggingServerHttpResponse.java:338]: maybe you can refactor this code?

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/datamask/KeyWordMatch.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    private Pattern p;
+
+    /**
+     * generate regex.
+     *
+     * @param keyWordSet keyWord set
+     */
+    public KeyWordMatch(final Set<String> keyWordSet) {
+
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/datamask/KeyWordMatch.java:36]: whitespace?

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/body/LoggingServerHttpResponse.java (hunk 1 above)
Comment (wanghuan0606) [reply shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-common/src/main/java/org/apache/shenyu/plugin/logging/common/body/LoggingServerHttpResponse.java:338]: ok

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-console/pom.xml [hunk 3]
```diff
@@ -32,6 +32,11 @@
             <artifactId>shenyu-plugin-base</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.shenyu</groupId>
+            <artifactId>shenyu-plugin-logging-common</artifactId>
+            <version>${project.version}</version>
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-console/pom.xml:38]: why use this?

**Comment by moremind** [comment]
pls resolve conflicts

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-console/pom.xml (hunk 3 above)
Comment (wanghuan0606) [reply shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-console/pom.xml:38]: reuse DataMaskByCharReplace、DataMaskByMD5、DataMaskInterface、KeyWordMatch

**Pull request closed without merging**